spatial/r3/icosahedron_example_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3_test
+
+import (
+	"fmt"
+
+	"gonum.org/v1/gonum/spatial/r3"
+)
+
+// edgeIdx represents an edge of the icosahedron
+type edgeIdx [2]int
+
+// subdivideEdge takes the vertices list and indices first and second to
+// the vertices defining the edge that will be subdivided. lookup is a
+// table of all newly generated vertices from previous calls to subdivideEdge
+// so as to not duplicate vertices.
+func subdivideEdge(lookup map[edgeIdx]int, vertices []r3.Vec, first, second int) (int, []r3.Vec) {
```

Comment:
I reversed the order of the functions but left them before the example following suit of the euler angle example. Should I add them after the example or is this OK?

Reply:
The Euler example is a special case; it was in that order because people know (or think they know) what Euler Angles are and want to use them, and it was written as a warning. This example has helpers that are not (AFAICS) something to be copy-pasted into their code.